src/server/mappers/TagMapper.ts

```diff
@@ -0,0 +1,22 @@
+/* eslint-disable class-methods-use-this, lines-between-class-members, no-dupe-class-members */
+import { injectable } from 'inversify'
+import { StorableTag } from '../repositories/types'
+import { TagType } from '../models/tag'
+import { Mapper } from './Mapper'
+
+@injectable()
+export class TagMapper implements Mapper<StorableTag, TagType> {
+  persistenceToDto(tagType: TagType): StorableTag
+  persistenceToDto(tagType: TagType | null): StorableTag | null {
+    if (!tagType) {
+      return null
+    }
+    return {
+      id: tagType.id,
+      tagKey: tagType.tagKey,
+      tagString: tagType.tagString,
+    }
+  }
+}
```

Comment:
Wondering what the purpose of the `Mapper` class is since we're not permuting the object.

Here, we're only checking for existence and returning the object as is if it exists, otherwise we return null - the same can be achieved without this `Mapper` layer since the object retrieved from the DB would have the same attributes/ return null too otherwise.

Reply:
Generally, it's a good idea to separate Model object and client facing object since they can be quite different.